catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/ocr/OcrShowTextBrickTest.java

```diff
@@ -0,0 +1,267 @@
+
+		AssetManager assetManager = InstrumentationRegistry.getContext().getResources().getAssets();
+		String[] files = assetManager.list(argAssetDir);
+
+		for (String file : files) {
+
+			String absAssetFilePath = addTrailingSlash(argAssetDir) + file;
+			String[] subFiles = assetManager.list(absAssetFilePath);
+
+			if (subFiles.length == 0) {
+				// It is a file
+				String destFilePath = addTrailingSlash(destDirPath) + file;
+				copyAssetFile(absAssetFilePath, destFilePath);
+			} else {
+				// It is a sub directory
+				copyDirOrFileFromAssetManager(absAssetFilePath, addTrailingSlash(argDestinationDir) + file);
+			}
+		}
+
+		return destDirPath;
+	}
+
+	private static void copyAssetFile(String assetFilePath, String destinationFilePath) throws IOException {
+		InputStream in = InstrumentationRegistry.getContext().getResources().getAssets().open(assetFilePath);
+		OutputStream out = new FileOutputStream(destinationFilePath);
+
+		byte[] buf = new byte[1024];
+		int len;
+		while ((len = in.read(buf)) > 0) {
+			out.write(buf, 0, len);
+		}
+		in.close();
+		out.close();
+	}
+
+	private static String addTrailingSlash(String path) {
+		if (path.charAt(path.length() - 1) != '/') {
+			path += "/";
+		}
+		return path;
+	}
+
+	private static String addLeadingSlash(String path) {
+		if (path.charAt(0) != '/') {
+			path = "/" + path;
+		}
+		return path;
+	}
+
+	private static void createDir(File dir) throws IOException {
+		if (dir.exists()) {
+			if (!dir.isDirectory()) {
+				throw new IOException("Can't create directory, a file is in the way");
+			}
+		} else {
+			dir.mkdirs();
+			if (!dir.isDirectory()) {
+				throw new IOException("Unable to create directory");
+			}
+		}
+	}
+
+	public static boolean deleteDirectory(File path) {
+		if (path.exists()) {
+			File[] files = path.listFiles();
+			for (File file : files) {
+				if (file.isDirectory()) {
+					deleteDirectory(file);
+				} else {
+					file.delete();
+				}
+			}
+		}
+		return path.delete();
+	}
+
+	@Test
+	public void testShowHindiNumberFormatAndArabicString() throws Exception {
+		createProject(HI_HINDI_NF);
+		baseActivityTestRule.launchActivity(null);
+
+		onView(withId(R.id.button_play))
+				.perform(click());
+		onView(isRoot())
+				.perform(CustomActions.wait(500));
+
+		Bitmap bitmap = Screenshot.capture().getBitmap();
+		TessOCR tessocr = new TessOCR("ara");
+		final String srcText = tessocr.getOCRResult(bitmap);
+
+		assertFalse("stage screenshot contains a text which is :" + srcText, srcText.isEmpty());
+
+		for (int i = 0; i < srcText.length(); i++) {
+			String str = String.valueOf(srcText.charAt(i));
+			assertTrue(str + " does not belong to arAlphabetAndNumbers Array", Arrays.asList(arAlphabetAndNumbers).contains(str));
+		}
+
+		tessocr.onDestroy();
+	}
+
+	@Test
+	public void testShowHindiNumberFormatAsEngNumberFormat() throws Exception {
+		createProject(AR_ONE_TO_ZERO);
+		baseActivityTestRule.launchActivity(null);
+
+		onView(withId(R.id.button_play))
```

Comment:
use StageActivity in the test rule, and skip the wait if possible. (if it fails with the wait, please have a look at eg. SayBubbleBrickStageTest, ive wrote a "ScriptEvaluationGateBrick" that helps you ensure that your brick under test has been executed in stage before the test continues.